### PR TITLE
Add Assembla support

### PR DESCRIPTION
**Pull request by ErikMediaMonks**
### New Pull Request Checklist

- [x] Run `go fmt` on your files (e.g. `go fmt ./service/common.go`, or on the whole `service` folder: `go fmt ./service/...`)
- [x] Write tests for your code
  - The tests should cover both "success" and "error" cases.
  - The tests should also check **all the returned variables**, don't ignore any returned value!
  - Ideally the tests should be easily readable, we usually use tests to document our code instead of code comments.
    An example, if you'd write a comment like "Given X this function will return Y" or
    "Beware, if the input is X this function will return Y" then you should implement this as
    a unit test, instead of writing it as a comment.
- [x] If your Pull Request is more than a bug fix you should also check `README.md` and change/add the descriptions there - also
  feel free to add yourself as a contributor if you implement support for a new service ;)
- [x] Before creating the Pull Request you should also run `bitrise run test` with the [Bitrise CLI](https://www.bitrise.io/cli),
  to perform all the automatic checks (which will run on your Pull Request when you open it).

### Summary of Pull Request

This Pull Request will add Atlassian Assembla Webook support!

### CODE REVIEW

**Comment by viktorbenei** [comment]
Hi @ErikMediaMonks ,

Sorry for the long delay, due to vacation time we have a significant backlog of PRs/reviews, but I'll try to review your PR as soon as possible - scheduled it for tomorrow.

P.S.: after a quick "scan through" it seems 👍 but I'll check it a bit more thoroughly tomorrow :)

**Comment by viktorbenei** [comment]
Still a bit behind schedule, but it's at the top of my list for Monday! @ErikMediaMonks

File: service/hook/assembla/assembla.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	Title   string	`json:"title"`
+	Body    string	`json:"body"`
+	Author	string	`json:"author"`
+}
+
+// GitEventModel ...
+type GitEventModel struct {
+	RepositorySuffix	string      `json:"repository_suffix"`
+	RepositoryURL		string      `json:"repository_url"`
+	Branch				string      `json:"branch"`
```
Comment (viktorbenei) [inline service/hook/assembla/assembla.go]: indentation fix

File: service/hook/assembla/assembla.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+		return hookCommon.TransformResultModel{
+			Error: fmt.Errorf("Git branch can't be empty"),
+		}
+	}
+
+	triggerAPIParams := []bitriseapi.TriggerAPIParamsModel{
+		{
+			BuildParams: bitriseapi.BuildParamsModel{
+				CommitMessage: pushEvent.MessageEventModel.Body,
+				Branch:        pushEvent.GitEventModel.Branch,
```
Comment (viktorbenei) [inline service/hook/assembla/assembla.go]: Use `CommitID` to specify the commit hash! E.g.: https://github.com/bitrise-io/bitrise-webhooks/blob/master/service/hook/github/github.go#L114

Without that the build will build the tip of the branch, and a rebuild of the build will also build the tip of the branch (instead of the same commit hash)!

This means that e.g. if you push a commit to the `master` branch, that starts a build on bitrise.io, then you push more code to `master` and then you try to rebuild a previous build that won't build the same commit hash, that will build the latest version of `master`. The intended behaviour of these webhook integrations is to include all the required infos for a rebuild, so that a rebuild actually re-builds the exact same state of the repository.

File: service/hook/assembla/assembla_test.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+			hookTransformResult := transformPushEvent(pushEvent)
+			err := detectAssemblaData(pushEvent)
+			require.Equal(t, err, nil)
+			require.NoError(t, hookTransformResult.Error)
+			require.False(t, hookTransformResult.ShouldSkip)
+			require.Equal(t, []bitriseapi.TriggerAPIParamsModel{
+				{
+					BuildParams: bitriseapi.BuildParamsModel{
+						CommitMessage: "ErikPoort pushed 1 commits [branchname]\n",
+						Branch:        "branchname",
```
Comment (viktorbenei) [inline service/hook/assembla/assembla_test.go]: should also include:

```
CommitHash: "sha1chars11",
```

File: service/hook/assembla/assembla_test.go [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+					},
+					TriggeredBy: "ErikPoort",
```
Comment (viktorbenei) [inline service/hook/assembla/assembla_test.go]: `TriggeredBy` should be `"webhook"` right now. We'll add "author" infos a bit later, right now the `TriggeredBy` indicates the source *type* (webhook, manual, curl, ...) not the exact author.

If you're interested feel free to create a feature request for tracking the status of the "author" info at https://discuss.bitrise.io/c/feature-request so that we can keep you up to date with the progress ;)

File: service/hook/assembla/assembla.go [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+			},
+			TriggeredBy: pushEvent.MessageEventModel.Author,
```
Comment (viktorbenei) [inline service/hook/assembla/assembla.go]: `TriggeredBy` should be `"webhook"` right now. We'll add "author" infos a bit later, right now the `TriggeredBy` indicates the source *type* (webhook, manual, curl, ...) not the exact author.

If you're interested feel free to create a feature request for tracking the status of the "author" info at https://discuss.bitrise.io/c/feature-request so that we can keep you up to date with the progress ;)

File: service/hook/assembla/assembla.go [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+	contentType := header.Get("Content-Type")
+	if contentType == "" {
+		return "", errors.New("No Content-Type Header found")
+	}
+
+	return contentType, nil
+}
+
+func detectAssemblaData(pushEvent PushEventModel) error {
+	if (pushEvent.GitEventModel.CommitID == "---") ||
```
Comment (viktorbenei) [inline service/hook/assembla/assembla.go]: I'd also suggest you to check whether these are empty (e.g. `pushEvent.GitEventModel.CommitID == ""`) in addition to the `"---"` value

File: service/hook/assembla/assembla_test.go [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+				Action: "committed",
+				Object: "Changeset",
+			},
+			MessageEventModel: MessageEventModel{
+				Title: "1 commits [branchname]",
+				Body: "ErikPoort pushed 1 commits [branchname]\n",
+				Author: "ErikPoort",
+			},
+			GitEventModel: GitEventModel{
+				RepositorySuffix: "---",
```
Comment (viktorbenei) [inline service/hook/assembla/assembla_test.go]: add a test where these are not even specified (empty string, the default if you don't specify it)

**Comment by viktorbenei** [comment]
Let me know if I can help with anything, and thanks for the pull request! :)

**Comment by ErikMediaMonks** [comment]
Hey, I have made the changes requested!

**Comment by viktorbenei** [comment]
[Released in v1.1.28](https://github.com/bitrise-io/bitrise-webhooks/releases/tag/1.1.28) - thanks again @ErikMediaMonks ! :)

**Comment by ErikMediaMonks** [comment]
No problem!
Can you give me an ETA on when we can use this on bitrise.io? 
(Or do I just need to wait for cache to invalidate?)

**Comment by viktorbenei** [comment]
You can now use it, following https://github.com/bitrise-io/bitrise-webhooks/blob/master/README.md#assembla---setup--usage

The UI will be updated soon, in the meantime probably the easiest is to copy paste the GitHub hook url from the code tab and replace github in the url with assembla

If you need any help just let me know! 😉
